client/pluginmanager/csimanager/usage_tracker.go

```diff
@@ -0,0 +1,70 @@
+package csimanager
+
+import (
+	"sync"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// volumeUsageTracker tracks the allocations that depend on a given volume
+type volumeUsageTracker struct {
+	state   map[volumeUsageKey][]*structs.Allocation
```

Comment:
Because we're copying this struct in `appendAlloc` anyways, does it make more sense to track the alloc ID rather than the pointer to the struct?